WG_GO_CORE.md

```diff
@@ -0,0 +1,177 @@
+   * Document and highlight best practices for dealing with the common hurdles in moving to immutable-first: e.g. “save packages in a merkle tree; build a package index merkle tree; use IPNS to handle updates of the tip of package index tree”.
+* `M (P1)`: There is a way to prioritize different peers to optimize network utilization
+   * Bitswap peer prioritization supports pinning local / responsive peers 
+
+
+### 🗂 Large Files
+
+Go-IPFS is fast, efficient and stable when handling and transferring large amounts of data. (“large files” = 10GB - 1TB, “many small files” = 10K-1M 1MB files). The goals in this section are exclusive of finding the data (DHT/providers etc.). This is just the speed of transferring data when already connected to IPFS nodes that have that data.
+
+* `M (P0)`: Awesome go-ipfs benchmark test suite exists comparing ipfs performance and transfer size relative to bittorrent, rsync, http, cp, dd
+* `M (P0)`: Go-IPFS bitswap (or equivalent) can transfer large files from many peers at 0.8 times the speed of BitTorrent. 
+   * Bitswap session improvements that improve transfer efficiency by reducing duplicate blocks and efficiently fetching related data from many peers simultaneously. 
+   * Graphsync selector implementation that can understand non-overlapping subsets of data to select from multiple hosts and the driver that can build up graphsync requests for many peers to maximize efficiency and throughput. 
+* `M (P0)`: Go-IPFS can transfer many small files at 0.8 times the speed of rsync (for single peers) or BitTorrent (for many peers)
+   * Bitswap session improvements
+   * Graph Exchange that can route requests efficiently
+   * Graphsync
+   * Flexible provider strategies
+* `M (P1)`: Go-IPFS can transfer large files from a single peer at 0.8 times the speed of HTTP. 
+   * Flexible provider strategies to use the DHT efficiently (and therefore not slow IPFS down while adding/fetching files) while still preserving the reachability of data. 
+* `M (P1)`: Go-IPFS can add a directory of many small files at 50% the speed of a simple local hard drive copy (or dd) (10k x 1MB files)
+   * Flexible provider strategies to limit the number of DHT provider advertisements we make
+   * Faster datastore (Badger or equivalent) that doesn’t slow down as the number of blocks stored grows large
+   * GC improvements (transactions, speed)
+* `M (P2)`: Data transferred over the network is no more than 1.2x the size of the data being transferred 
+   * Bitswap session improvements to reduce duplicates
+   * Graphsync
+
+
+### 🔄 Decentralized Web
+
+go-ipfs is the go-to backend for building DApps.
+
+* `M (P0)`: Provider lookups are efficient (<1 sec) in real-world network conditions (ex between physically distant peers)
+* `M (P0)`: ipfs://, ipns:// work in web browsers
+   * Base32 CIDs by default
+   * Base32 IPNS keys (or make IPNS keys CIDs)

```

Comment:
This is related to a conversation happening in Libp2p (though could be separable). I think browser folks really want this, but not sure we have bandwidth on it this quarter, so maybe Q2 once we have the Base32 CIDs? @kevina @Stebalien for your thoughts...
```suggestion
   * Q2 - Base32 IPNS keys (or make IPNS keys CIDs)
```

Reply:
This is related to that discussion but yes, it probably won't get finalized until Q2. It's not hard exactly, it just requires consensus.